pkg/apis/options/providers.go

```diff
@@ -189,6 +189,10 @@ type GoogleOptions struct {
 	AdminEmail string `json:"adminEmail,omitempty"`
 	// ServiceAccountJSON is the path to the service account json credentials
 	ServiceAccountJSON string `json:"serviceAccountJson,omitempty"`
+	// UseApplicationDefaultCredentials is a boolean whether to use Application Default Credentials instead of a ServiceAccountJSON
+	UseApplicationDefaultCredentials bool `json:"useApplicationDefaultCredentials,omitempty"`
```

Comment:
Given the alpha status of this new config struct, how would you do this if you had the choice by default? Have you considered that the credentials should be a discriminated union?

I guess it's an error for ServiceAccountJSON to be set at the same time as UseApplicationDefaultCredentials?

Reply:
I'm not sure how to do this, if you could provide an example it'd be greatly appreciated. In the meantime I'll look for an example to do it myself.

Reply:
After everything has been moved to the new config structure we could maybe think about a better / new validation logic for parsing the config including discriminated unions for all providers in general. Unfortunately, this is not as easily implemented in go as in typescript.